optimum/neuron/trainers.py

```diff
@@ -182,6 +183,8 @@ def __init__(self, *args, **kwargs):
         # Make the model Neuron-compatible for generation.
         patch_generation_mixin_to_neuron_generation_mixin(self.model)
 
+        set_neuron_cc_optlevel_for_model(self.model, optlevel=self.args.neuron_cc_optlevel)
```

Comment:
I don't see the benefit of adding yet another parameter that is more or less a duplicate of the compiler optlevel: if the llama models require optlevel 1 because compilation takes too long, then we put it somewhere in the guidelines and ask the user to export it accordingly.
This adds complexity, which will increase the maintenance cost and the test coverage, but it does not fix the real issue (llama compilation takes forever).

Reply:
The idea is to include this "knowledge" so that the user does not have to care about that.
For instance here: #313 .

I do not think this will add too much maintenance code, it is basically a function that sets a flag in the compiler. 

I think it's a nicer API for the end user to be able to specify the opt level, if they want to do that, as an argument instead of using the environment variable.

Reply:
I think it is a bit superfluous, but agree to disagree.